warehouse/forklift/legacy.py

```diff
@@ -1169,28 +1200,33 @@ def file_upload(request):
                     "requires_python",
                 }
             },
-            uploader=request.user,
+            uploader=request.user if request.user else None,
             uploaded_via=request.user_agent,
         )
         request.db.add(release)
-        # TODO: This should be handled by some sort of database trigger or
-        #       a SQLAlchemy hook or the like instead of doing it inline in
-        #       this view.
-        request.db.add(
-            JournalEntry(
-                name=release.project.name,
-                version=release.version,
-                action="new release",
-                submitted_by=request.user,
-                submitted_from=request.remote_addr,
+
+        if request.user:
+            # TODO: This should be handled by some sort of database trigger or
+            #       a SQLAlchemy hook or the like instead of doing it inline in
+            #       this view.
+            request.db.add(
+                JournalEntry(
+                    name=release.project.name,
+                    version=release.version,
+                    action="new release",
+                    submitted_by=request.user,
+                    submitted_from=request.remote_addr,
+                )
             )
-        )
 
         project.record_event(
             tag="project:release:add",
             ip_address=request.remote_addr,
             additional={
-                "submitted_by": request.user.username,
+                "ephemeral": request.user is None,
```

Comment:
Flagging: It might make sense for this to be `nonuser: bool` instead, since `request.user is None` might not always imply an OIDC ephemeral token in the future.